Commit a34fbf7 breaks compilation

##### System information (version)
- OpenCV => Latest
- Operating System / Platform => Ubuntu 16.04
- Compiler => gcc 5.4
I am also on CUDA 8.

##### Detailed description

If I compile the latest code I will get compilation error as shown below

``````
[ 44%] Building NVCC (Device) object modules/cudaimgproc/CMakeFiles/cuda_compile.dir/src/cuda/cuda_compile_generated_clahe.cu.o
/home/jasjuang/opencv/modules/core/include/opencv2/core/cuda/reduce.hpp(61): error: name followed by "::" must be a class or namespace name

/home/jasjuang/opencv/modules/core/include/opencv2/core/cuda/reduce.hpp(61): error: name followed by "::" must be a class or namespace name

/home/jasjuang/opencv/modules/core/include/opencv2/core/cuda/reduce.hpp(61): error: expected an identifier

/home/jasjuang/opencv/modules/core/include/opencv2/core/cuda/reduce.hpp(61): error: expected a ";"

4 errors detected in the compilation of "/tmp/tmpxft_0000464c_00000000-15_clahe.compute_60.cpp1.ii".
CMake Error at cuda_compile_generated_clahe.cu.o.cmake:264 (message):
  Error generating file
  /home/jasjuang/opencv/build/modules/cudaimgproc/CMakeFiles/cuda_compile.dir/src/cuda/./cuda_compile_generated_clahe.cu.o


modules/cudaimgproc/CMakeFiles/opencv_cudaimgproc.dir/build.make:1112: recipe for target 'modules/cudaimgproc/CMakeFiles/cuda_compile.dir/src/cuda/cuda_compile_generated_clahe.cu.o' failed
make[2]: *** [modules/cudaimgproc/CMakeFiles/cuda_compile.dir/src/cuda/cuda_compile_generated_clahe.cu.o] Error 1
CMakeFiles/Makefile2:6060: recipe for target 'modules/cudaimgproc/CMakeFiles/opencv_cudaimgproc.dir/all' failed
make[1]: *** [modules/cudaimgproc/CMakeFiles/opencv_cudaimgproc.dir/all] Error 2
Makefile:160: recipe for target 'all' failed
make: *** [all] Error 2
``````

If I do a `git checkout 6328076`, everything compiles as usual.

--- a/modules/core/include/opencv2/core/cuda/color.hpp
+++ b/modules/core/include/opencv2/core/cuda/color.hpp
@@ -306,4 +306,4 @@ namespace cv { namespace cuda { namespace device
 
 //! @endcond
 
-#endif // OPENCV_CUDA_BORDER_INTERPOLATE_HPP
+#endif // OPENCV_CUDA_COLOR_HPP

--- a/modules/core/include/opencv2/core/cuda/detail/reduce_key_val.hpp
+++ b/modules/core/include/opencv2/core/cuda/detail/reduce_key_val.hpp
@@ -40,8 +40,8 @@
 //
 //M*/
 
-#ifndef OPENCV_CUDA_PRED_VAL_REDUCE_DETAIL_HPP_
-#define OPENCV_CUDA_PRED_VAL_REDUCE_DETAIL_HPP_
+#ifndef OPENCV_CUDA_PRED_VAL_REDUCE_DETAIL_HPP
+#define OPENCV_CUDA_PRED_VAL_REDUCE_DETAIL_HPP
 
 #include <thrust/tuple.h>
 #include "../warp.hpp"
@@ -499,4 +499,4 @@ namespace cv { namespace cuda { namespace device
 
 //! @endcond
 
-#endif // OPENCV_CUDA_PRED_VAL_REDUCE_DETAIL_HPP_
+#endif // OPENCV_CUDA_PRED_VAL_REDUCE_DETAIL_HPP

--- a/modules/core/include/opencv2/core/cuda/reduce.hpp
+++ b/modules/core/include/opencv2/core/cuda/reduce.hpp
@@ -202,4 +202,4 @@ namespace cv { namespace cuda { namespace device
 
 //! @endcond
 
-#endif // OPENCV_CUDA_UTILITY_HPP
+#endif // OPENCV_CUDA_REDUCE_HPP

--- a/modules/core/include/opencv2/core/cuda/type_traits.hpp
+++ b/modules/core/include/opencv2/core/cuda/type_traits.hpp
@@ -40,8 +40,8 @@
 //
 //M*/
 
-#ifndef OPENCV_CUDA_REDUCE_DETAIL_HPP
-#define OPENCV_CUDA_REDUCE_DETAIL_HPP
+#ifndef OPENCV_CUDA_TYPE_TRAITS_HPP
+#define OPENCV_CUDA_TYPE_TRAITS_HPP
 
 #include "detail/type_traits_detail.hpp"
 
@@ -87,4 +87,4 @@ namespace cv { namespace cuda { namespace device
 
 //! @endcond
 
-#endif // OPENCV_CUDA_REDUCE_DETAIL_HPP
+#endif // OPENCV_CUDA_TYPE_TRAITS_HPP

--- a/modules/core/include/opencv2/core/private.cuda.hpp
+++ b/modules/core/include/opencv2/core/private.cuda.hpp
@@ -169,4 +169,4 @@ namespace cv { namespace cuda
 
 //! @endcond
 
-#endif // OPENCV_CORE_CUDA_PRIVATE_HPP
+#endif // OPENCV_CORE_PRIVATE_CUDA_HPP

--- a/modules/cudafilters/include/opencv2/cudafilters.hpp
+++ b/modules/cudafilters/include/opencv2/cudafilters.hpp
@@ -328,4 +328,4 @@ CV_EXPORTS Ptr<Filter> createMedianFilter(int srcType, int windowSize, int parti
 
 }} // namespace cv { namespace cuda {
 
-#endif /*OPENCV_CUDAFILTERS_HPP */
+#endif /* OPENCV_CUDAFILTERS_HPP */

